joomla/joomla-cms
# [5.0][Events] Finder event classes

### Summary of Changes
Implementing Event classes for Finder events.
The pr depend from https://github.com/joomla/joomla-cms/pull/41226

Affected events:

- [x] onFinderCategoryChangeState
- [x] onFinderChangeState
- [x] onFinderAfterDelete
- [x] onFinderBeforeSave
- [x] onFinderAfterSave
- [x] onFinderResult
- [x] onPrepareFinderContent

### Testing Instructions
Apply patch.
Make sure finder plugins is enabled.
Try edit content, run indexer, use smart search.


### Actual result BEFORE applying this Pull Request
Works


### Expected result AFTER applying this Pull Request
Works


### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: https://docs.joomla.org/Plugin/Events#Finder
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/177
